syntax = "proto3";
package echo;

import "google/protobuf/timestamp.proto";

message EchoRequest{
    string message = 1;
}

message EchoResponse{
    string message = 1;
}

message EchoLockResponse{
    int64 id = 1;
    string message = 2;
    google.protobuf.Timestamp start = 3;
    google.protobuf.Timestamp end = 4;
}



service Echo {
    rpc UnaryEcho( EchoRequest ) returns (EchoResponse) {}
    rpc UnaryLock (EchoRequest) returns (EchoLockResponse) {}
}